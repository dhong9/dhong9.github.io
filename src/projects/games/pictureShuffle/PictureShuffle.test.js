import renderer from "react-test-renderer";
import PictureShuffle from "projects/games/pictureShuffle";
import React from "react";

let realUseContext;
let useContextMock;
// Setup mock
beforeEach(() => {
    realUseContext = React.useContext;
    useContextMock = React.useContext = jest.fn();
});
// Cleanup mock
afterEach(() => {
    React.useContext = realUseContext;
});


// Define Mocks
jest.mock("components/MKButton", () => {
    const { forwardRef } = jest.requireActual("react");
    return {
        __esModule: true,
        default: forwardRef(() => <button>Mock Button</button>)
    };
});
jest.mock("components/MKBox", () => {
    const { forwardRef } = jest.requireActual("react");
    return {
        __esModule: true,
        default: forwardRef(() => <div>Mock Box</div>)
    };
});
jest.mock("layouts/sections/components/BaseLayout", () => {
    const { forwardRef } = jest.requireActual("react");
    return {
        __esModule: true,
        default: forwardRef(() => <div>Mock Layout</div>)
    };
});
jest.mock("layouts/sections/components/View", () => {
    const { forwardRef } = jest.requireActual("react");
    return {
        __esModule: true,
        default: forwardRef(() => <div>Mock View</div>)
    };
});
jest.mock("react-monaco-editor", () => {
    const { forwardRef } = jest.requireActual("react");
    return {
        __esModule: true,
        default: forwardRef(() => <div>Mock Editor</div>)
    };
});
jest.mock("context/AuthContext", () => {
    const { createContext } = jest.requireActual('react'); 
    return {
        __esModule: true,
        default: createContext()
    }
});

describe("PictureShuffle", () => {
    it("renders", () => {
        useContextMock.mockReturnValue("Test Value");
        const component = renderer.create(<PictureShuffle />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});