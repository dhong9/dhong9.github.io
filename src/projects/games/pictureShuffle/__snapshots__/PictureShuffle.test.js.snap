// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PictureShuffle renders 1`] = `
<div>
  Mock Layout
</div>
`;
